crawlerr/services/crawler_service.py

```
@@ -442,6 +393,377 @@ async def get_sitemap_urls(self, base_url: str) -> List[str]:
+                    else:
+                        stats.total_pages_failed += 1
+                        logger.warning(f"Failed to crawl {current_url}: {result.error_message}")
+                        
+                except Exception as e:
+                    stats.total_pages_failed += 1
+                    logger.error(f"Error crawling {current_url}: {e}")
+                    
+        logger.info(f"Recursive crawl discovered {len(pages)} pages across {max_depth} depth levels")
+        return pages
+    
+    async def _batch_crawl(self, urls: List[str], stats: CrawlStatistics) -> List[PageContent]:
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)